logstash.go

```diff
@@ -30,8 +41,86 @@ func Init(logLevel string, logFileName string, env string, service string) bool
 	})
 
 	log.SetLevel(LogLevels[logLevel])
-	mw := io.MultiWriter(os.Stdout, logFile)
-
-	log.SetOutput(mw)
+	setLogFile(logFile)
+	go rotate(logFileName)
 	return true
 }
+
+func setLogFile(writer io.Writer) {
+	log.SetOutput(io.MultiWriter(os.Stdout, writer))
+}
+
+// rotate checks periodically if the
+// when the current logfile
+func rotate(logFileName string) {
+	log.Debug("starting rotation for", logFileName)
+
+	if stopRotation != nil {
+		log.Debug("stopping rotation")
+		close(stopRotation)
+	}
+	stopRotation = make(chan struct{})
+
+	ticker := time.NewTicker(rotationCheckInterval)
+	for {
+		select {
+		case <-stopRotation:
+			log.Debug("stopping rotation for", logFileName)
+			ticker.Stop()
+			return
+		case <-ticker.C:
+			f, err := os.Stat(logFileName)
+			if f.Size() > maxLogFileSize {
+				log.Debugf("log file too large, rotating the log file")
+				rotatedFilename := logFileName + ".1"
```

Comment:
with a static postfix, would the next rotation overwrite the first one?

Reply:
if you look after, before renaming the current log file to this, I try to detect this file, rename it and compress it (the compression happens asyncronously).